modern-publish-plugin.md

```diff
@@ -0,0 +1,211 @@
+  # Without this the arifact id would always be based on `project.getName()`.
+  # Implementation in Kotlin MPP is kind of hacky using replace so it might be
+  # best to just not support it. How do we handle compatibility?
+  baseArtifactId = "library"
+
+  # Same API as Gradle's, can probably just use their interfaces/classes
+  # Will be applied to all publications
+  pom {
+    name = "My Library"
+    description = "A description of what my library does."
+    inceptionYear = "2020"
+    url = "https://github.com/username/mylibrary/"
+    
+    scm {
+        connection = "scm:git:git://github.com/username/mylibrary.git"
+        developerConnection = "scm:git:ssh://git@github.com/username/mylibrary.git"
+        url = "https://github.com/username/mylibrary/"
+    } 
+    
+    licenses {
+        license {
+            name = "The Apache Software License, Version 2.0"
+            url = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+            distribution = "repo"
+        }
+    }
+    developers {
+        developer {
+            id = "username"
+            name = "User Name"
+            url = "https://github.com/username/"
+        }
+    }
+  }
+
+  # Configure closeAndReleaseRepository
+  nexus {
+     baseUrl = "https://your_nexus_instance"
+     stagingProfile = "net.example"
+     # Start supporting the Gradle credentials API
+     respositoryUserName = "username"
+     respositoryPassword = "password"
+  }
+}
+```
+
+#### `com.vanniktech.maven.publish.java`
+
+Meant to be used with `java`, `java-library`, `org.jetbrains.kotlin.jvm`. 
+
+Pure Gradle configuration:
+```
+publishing {
+    publications {
+        create<MavenPublication>("maven") {
+            from(components["java"])
+        }
+    }
+}
+
+java {
+  withSourcesJar()
+  withJavadocJar()
+}
+```
+
+The plugin doesn't need to offer anything for sources and javadocs jars. It will
+simply create a publication using `components["java"]`.
+
+#### `com.vanniktech.maven.publish.gradle`
+
+Meant to be used with `java-gradle-plugin`.
+
+Pure Gradle configuration:
+```
+java {
+  withSourcesJar()
+  withJavadocJar()
+}
+```
+
+Gradle automatically adds a publication when both `java-gradle-plugin` and 
+`maven-publish` are applied. There is no need for our own APIs, so this plugin
+does not have any functionality of it's own.
+
+#### `com.vanniktech.maven.publish.android`
+
+Meant to be used with `com.android.library`.
+
+Pure Gradle configuration:
+```
+afterEvaluate {
+    publishing {
+        publications {
+            create<MavenPublication>("maven") {
+                from(components["release"])
+            }
+        }
+    }
+}
+```
+
+
+There is no API available to create source and javadoc jars for Android 
+projects. The plugin will provide an API to configure those as well as 
+which variant to publish. These properties/methods can be removed/deprecated once
+AGP offers it's own methods and adds multi variant publishing.
+
+```
+androidPublishCompat {
```

Comment:
Better name?

Reply:
Can't we use androidPublish?

Reply:
Sure, the main reason why I added was compat was because it's meant to be temporary until AGP handles this on it's own.